sdk/search/Azure.Search.Documents/src/Options/SearchOptions.cs

```diff
@@ -207,62 +205,59 @@ internal string QueryAnswerRaw
 
                 if (QueryAnswer.HasValue)
                 {
-                    queryAnswerStringValue = $"{QueryAnswer.Value}{QueryAnswerCountRawSplitter}{QueryAnswerCount.GetValueOrDefault(1)}{QueryAnswerThresholdRawSplitter}{QueryAnswerThreshold.GetValueOrDefault(0.7)}";
+                    var queryAnswerValues = new List<string>();
+
+                    if (QueryAnswerCount.HasValue)
+                    {
+                        queryAnswerValues.Add($"{QueryAnswerCountRaw}{QueryAnswerCount}");
+                    }
+
+                    if (QueryAnswerThreshold.HasValue)
+                    {
+                        queryAnswerValues.Add($"{QueryAnswerThresholdRaw}{QueryAnswerThreshold}");
+                    }
+
+                    queryAnswerStringValue = $"{QueryAnswer.Value}" + (queryAnswerValues.Count > 0 ? $"|{string.Join(",", queryAnswerValues)}" : "");
```

Comment:
Would you consider this a hot path? If so, there's a lot of extra allocations going on here. I recommend a `StringBuilder` with a simple marker to keep track of which token to prepend e.g.,

```csharp
StringBuilder queryAnswerStringValue = new(QueryAnswer.Value);

int tokens = 0;
char NextToken() => tokens switch {
  0 => '|',
  _ => ',',
};

if (QueryAnswerCount.HasValue)
{
  queryAnswerStringValue.Append(NextToken()).Append(QueryAnswerCount.Value);
}

if (QueryAnswerThreshold.Hasvalue)
{
  queryAnswerStringValue.Append(NextToken()).Append(QueryAnswerthreshold.Value);
}

return queryAnswerStringValue.ToString();
```

Reply:
Yes, also we need to set other int value after appending `QueryAnswerCount.Value`, otherwise the result would be "extractive|count-5|threshold-0.8".

```csharp
if (QueryAnswerCount.HasValue)
{
    queryAnswerStringValue.Append(NextToken()).Append($"{QueryAnswerCountRaw}{QueryAnswerCount.Value}");
    tokens = 1;
}
```

 I've updated the code. Thanks!

Reply:
I've updated the code. Thanks!